designer/client/src/components/graph/node-modal/editors/expression/FixedValuesEditor.tsx

```diff
@@ -34,39 +36,64 @@ function getOptions(
     }));
 }
 
-export default class FixedValuesEditor extends React.Component<Props> {
-    public static isSwitchableTo = (expressionObj: ExpressionObj, editorConfig) =>
-        editorConfig.possibleValues.map((v) => v.expression).includes(expressionObj.expression) || isEmpty(expressionObj.expression);
-    public static switchableToHint = () => "Switch to basic mode";
-    public static notSwitchableToHint = () => "Expression must be one of the predefined values to switch to basic mode";
-
-    currentOption(expressionObj: ExpressionObj, options: Option[]): Option {
+const FixedValuesEditorComponent = (props: Props) => {
+    const handleCurrentOption = (expressionObj: ExpressionObj, options: Option[]): Option => {
         return (
             (expressionObj && options.find((option) => option.value === expressionObj.expression)) || // current value with label taken from options
             (expressionObj && { value: expressionObj.expression, label: expressionObj.expression }) || // current value is no longer valid option? Show it anyway, let user know. Validation should take care
             null
         ); // just leave undefined and let the user explicitly select one
-    }
+    };
 
-    render() {
-        const { expressionObj, readOnly, onValueChange, className, showValidation, validators, editorConfig } = this.props;
-        const options = getOptions(editorConfig.possibleValues);
-        const currentOption = this.currentOption(expressionObj, options);
+    const { expressionObj, readOnly, onValueChange, className, showValidation, validators, editorConfig } = props;
+    const options = getOptions(editorConfig.possibleValues);
+    const currentOption = handleCurrentOption(expressionObj, options);
+    const theme = useTheme();
 
-        return (
-            <div className={`node-value-select ${className}`}>
-                <Creatable
-                    classNamePrefix={styles.nodeValueSelect}
-                    value={currentOption}
-                    onChange={(newValue) => onValueChange(newValue.value)}
-                    options={options}
-                    isDisabled={readOnly}
-                    formatCreateLabel={(x) => x}
-                    menuPortalTarget={document.body}
-                    createOptionPosition={"first"}
-                />
-                {showValidation && <ValidationLabels validators={validators} values={[currentOption.value]} />}
-            </div>
-        );
-    }
-}
+    const { control, input, valueContainer, singleValue, menuPortal, menu, menuList, menuOption } = selectStyled(theme);
+    return (
+        <div className={cx(className)}>
+            <Creatable
```

Comment:
Maybe we can create styled components?

Reply:
Unfortunately, no, because the dropdown menu is added to the body HTML element, so we need to set style as props here